lib/workflows/create_project.py

```diff
@@ -47,38 +80,201 @@ def create_project(
+    :param conn: OpenStack connection object
+    :type conn: Connection
+    :param project: OpenStack project object
+    :type project: Project
+    :param external_network: External Cloud network
+    :type external_network: Networks
+    :param number_of_floating_ips: Floating IP quota for project
+    :type number_of_floating_ips: int
+    :param number_of_security_group_rules: Security group quota for project
+    :type number_of_security_group_rules: int
+    """
+    network = create_network(
+        conn,
+        NetworkDetails(
+            name=f"{project.name}-network",
+            description="",
+            project_identifier=project.id,
+            provider_network_type=NetworkProviders.VXLAN,
+            port_security_enabled=True,
+            has_external_router=False,
+        ),
+    )
+
+    logger.info("Created network: %s", network.id)
+
+    external_rbac_policy = create_network_rbac(
+        conn,
+        NetworkRbac(
+            project_identifier=project.id,
+            network_identifier=external_network.value,
+            action=RbacNetworkActions.EXTERNAL,
+        ),
+    )
+
+    logger.info("Created external network rbac: %s", external_rbac_policy.id)
+
+    private_rbac_policy = create_network_rbac(
+        conn,
+        NetworkRbac(
+            project_identifier=project.id,
+            network_identifier=network.id,
+            action=RbacNetworkActions.SHARED,
+        ),
+    )
+
+    logger.info("Created private network rbac: %s", private_rbac_policy.id)
+
+    router = create_router(
+        conn,
+        RouterDetails(
+            router_name=f"{project.name}-router",
+            router_description="",
+            project_identifier=project.id,
+            external_gateway=external_network.value,
+            is_distributed=False,
+        ),
+    )
+
+    logger.info("Created router: %s", router.id)
+
+    subnet = create_subnet(
+        conn,
+        network_identifier=network.id,
+        subnet_name=f"{project.name}-subnet",
+        subnet_description="",
+        dhcp_enabled=True,
+    )
+
+    logger.info("Created subnet: %s", subnet.id)
+
+    add_interface_to_router(
+        conn,
+        project_identifier=project.id,
+        router_identifier=router.id,
+        subnet_identifier=subnet.id,
+    )
+
+    set_quota(
+        conn,
+        QuotaDetails(
+            project_identifier=project.id,
+            num_floating_ips=number_of_floating_ips,
+            num_security_group_rules=number_of_security_group_rules,
+        ),
+    )
+
+    # create default security group rules
+    if external_network == Networks.EXTERNAL:
+        create_external_security_group_rules(
+            conn, project_identifier=project.id, security_group_identifier="default"
+        )
+    elif external_network == Networks.JASMIN:
+        create_internal_security_group_rules(
+            conn, project_identifier=project.id, security_group_identifier="default"
         )
```

Comment:
I'm a bit nervous about this, it feels like a potential foot-gun where if we add a new internal sec group rule it implicitly will affect JASMIN because of an implementation detail

My gut says it's worth duplicating the method to make a JASMIN specific one that we/they can explicitly audit and verify even if it duplicates code